package/src/components/InPageMenuItem/v1/InPageMenuItem.js

```diff
@@ -0,0 +1,111 @@
+import { withComponents } from "@reactioncommerce/components-context";
+import { applyTheme, addTypographyStyles, preventAccidentalDoubleClick } from "../../../utils";
+
+const InPageMenuItemContainer = styled.div`
+  align-items: center;
+  ${(props) => {
+    if (props.isSelected) {
+      return `background-color: ${applyTheme("InPageMenuItemContainer.backgroundColor_selected")(props)};`;
+    }
+    return `background-color: ${applyTheme("InPageMenuItemContainer.backgroundColor_default")(props)};`;
+  }}
+  cursor: pointer;
+  display: flex;
+  flex-direction: row;
+  padding-left: ${applyTheme("InPageMenuItemContainer.paddingLeft")};
+  padding-right: ${applyTheme("InPageMenuItemContainer.paddingRight")};
+  padding-top: ${applyTheme("InPageMenuItemContainer.paddingTop")};
+  padding-bottom: ${applyTheme("InPageMenuItemContainer.paddingBottom")};
+  margin-bottom: ${applyTheme("InPageMenuItemContainer.marginBottom")};
+  `;
+const InPageMenuItemText = styled.div`
+  ${addTypographyStyles("InPageMenuItemText", "headingTextBold")}
+  display: flex;
+  flex: 1 1 auto;
+  `;
+const InPageMenuItemIcon = styled.span`
+  display: flex;
+  flex: 0 0 auto;
+  height: ${applyTheme("InPageMenuItemIcon.height")};
+  margin: 0;
+  width: ${applyTheme("InPageMenuItemIcon.width")};
+  svg {
+    height: ${applyTheme("InPageMenuItemIcon.height")};
+    transform: rotateZ(270deg);
+    width: ${applyTheme("InPageMenuItemIcon.width")};
+  }
+  `;
+
+class InPageMenuItem extends Component {
+  static propTypes = {
+    /**
+     * You can provide a `className` prop that will be applied to the outermost DOM element
+     * rendered by this component. We do not recommend using this for styling purposes, but
+     * it can be useful as a selector in some situations.
+     */
+    className: PropTypes.string,
+    /**
+     * If you've set up a components context using
+     * [@reactioncommerce/components-context](https://github.com/reactioncommerce/components-context)
+     * (recommended), then this prop will come from there automatically. If you have not
+     * set up a components context or you want to override one of the components in a
+     * single spot, you can pass in the components prop directly.
+     */
+    components: PropTypes.shape({
+      /**
+       * Pass either the Reaction iconExpand component or your own component that
+       * accepts compatible props.
+       */
+      iconExpand: PropTypes.node.isRequired,
+      Link: PropTypes.node.isRequired
+    }).isRequired,
+    /**
+     * URL to provide to MenuItem
+     */
+    href: PropTypes.string.isRequired,
+    /**
+     * Render selected style
+     */
+    isSelected: PropTypes.bool,
+    /**
+     * Menu Item label
+     */
+    label: PropTypes.node.isRequired,
+    /**
+     * Function to pass to button onClick
+     */
+    onClick: PropTypes.func
+  };
+
+  static defaultProps = {
+    isSelected: false
+  };
+
+  handleClick = preventAccidentalDoubleClick(() => {
```

Comment:
I think this method needs to do something like `event.preventDefault` or maybe the `href` doesn't need to be required. 

In the "With on click" example if you change the `href` prop to something like `href="/#!/thing"` then click the menu item the alert will fire but once it's closed the page will redirect to "/thing".